app/src/main/java/com/mk/tasky/agenda/presentation/detail/event/DetailEventViewModel.kt

```diff
@@ -0,0 +1,129 @@
+package com.mk.tasky.agenda.presentation.detail.event
+
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.setValue
+import androidx.lifecycle.SavedStateHandle
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.mk.tasky.agenda.presentation.home.HomeItemOptions
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.launch
+import java.time.LocalDateTime
+import javax.inject.Inject
+
+@HiltViewModel
+class DetailEventViewModel @Inject constructor(
+    savedStateHandle: SavedStateHandle
+) : ViewModel() {
+    var state by mutableStateOf(DetailEventState())
+        private set
+
+    init {
+        savedStateHandle.get<String>("date")?.let {
+            state = state.copy(
+                fromDate = LocalDateTime.parse(it).toLocalDate(),
+                fromTime = LocalDateTime.parse(it).toLocalTime(),
+                toDate = state.fromDate,
+                toTime = state.fromTime.plusMinutes(30)
+            )
+        }
+
+        val itemId = savedStateHandle.get<String>("id")
+        if (itemId != null) {
+            // TODO: Get Event from Repository
+            savedStateHandle.get<String>("action")?.let {
+                when (HomeItemOptions.from(it)) {
+                    HomeItemOptions.EDIT -> {
+                        state = state.copy(
+                            isEditing = true
+                        )
+                    }
+                    HomeItemOptions.OPEN -> {
+                        state = state.copy(
+                            isEditing = false
+                        )
+                    }
+                    HomeItemOptions.DELETE -> {
+                        viewModelScope.launch {
+                            // TODO: Delete
+                            state = state.copy(
+                                shouldExit = true
+                            )
+                        }
+                    }
+                    else -> Unit
+                }
+            }
+        }
+    }
+
+    fun onEvent(event: DetailEventEvents) {
+        when (event) {
+            DetailEventEvents.OnEdit -> {
+                state = state.copy(
+                    isEditing = true
+                )
+            }
+            DetailEventEvents.OnEventDelete -> {
+                state.id?.let {
+                    // TODO: Delete Event
+                }
+                state = state.copy(
+                    shouldExit = true
+                )
+            }
+            DetailEventEvents.OnNotificationReminderClick -> {
+                state = state.copy(
+                    showDropdown = true
```

Comment:
Personal preference, but I prefer calling the state variables like something that does not reveal what they will be used for on the UI layer, since you might use them for multiple things.

So here, it could be `isSelectingNotificationReminder` instead of `showDropdown` since that rather communicates what the user is doing instead of what's happening.

Or think about having a state `showProgressBar` instead of `isLoading`. Oftentimes, you want to disable a button if the state is loading, but if you call your state `showProgressBar` that doesn't fit well to disabling a button.

It's also valid to do it the other way around, but then I'd do it consistently (for example renaming `isEditing` to `shouldShowEditButton`)

Reply:
Awesome. Thanks for the tip!
Will rename it